### ISSUE TITLE
[Accessibility, a11y] Screen reader not reading out quick pick list item numbers correctly (macOS v12.3.1, VoiceOver)

### ISSUE THREAD

**Issue by MicroFish91** [microsoft/vscode#149395]
**VS Code Version**: 1.67.1

**Operating System**: macOS Monterey v12.3.1

**Screen Reader**: Apple VoiceOver

**Extensions Installed**:
Put together an extension to quickly demo the bare VS Code UI features without our normal SDK and extension wrappers.
"a11y-kit" (https://marketplace.visualstudio.com/items?itemName=MicroFish.accessibility-testkit)

**Repro Steps**:
  1. Install a11y-kit from VS Code Marketplace
  2. Open command palette and run a11y-kit's menu command
  3. Next, select: "Quick Picks: Show Quick Picks - Single"
  4. Highlight over the quick pick item options now shown below.

**Expected**:
Each quick pick item's list number should be read by screen reader when selected (i.e. "1 of 3", "2 of 3", "3 of 3").
Currently this is the behavior seen on NVDA but doesn't seem consistent on VoiceOver.

**Actual**:
The first list item's number is called out correctly as "1 of 3", but upon selecting the next quick pick item, the list number is no longer properly captured by the screen reader (the screen reader seems to detect the "2 of 3" for a split second but then clears to just the item field descriptors without calling out the list numbers).

**Attachments**:
![sqp_macos](https://user-images.githubusercontent.com/40250218/168168513-83e97b47-b3a8-4f1d-9914-168c114e3ff1.gif)

**Comment by fsteffi** [comment]
GitHubTags:#A11yMAS;#A11yTCS;#A11ySev2;#A11y_VSCode;#CAITestPass_May2020;#Visual Studio Code;#VoiceOver;#MAC;#WCAG1.3.1;#AINotInScope;

Primary bug:https://github.com/microsoft/vscode-azurefunctions/issues/3157

**Comment by isidorn** [comment]
@fsteffi this is a voice over issue. We set the correct `aria-posinset` on the items. 
Can you please open a new issue against webkit https://bugs.webkit.org/

I will also reach out to our VoiceOver contacts.

**Comment by bbarik06** [comment]
@isidorn Could you please provide the sample app or code pen. We have already logged the bug, but we need sample app for the external team so that they can produce a same control for them which isolates the problem and clarifies that the bug is in Apple's frameworks.

**Comment by isidorn** [comment]
VoiceOver team was kind enough to test this out and confirm this is fixed on MacOS 13 the current beta.

![Recording 2022-07-27 at 14 54 44](https://user-images.githubusercontent.com/1926584/181251832-8e1ccf36-3625-4fc1-8ef0-98d0348dc692.gif)

**Comment by isidorn** [comment]
After discussing with the accessibility testing team they confirmed that we should go ahead and close this bug. They will try to get the latest macOS to confirm that it is indeed fixed. Thus closing this as fixed in upstream.

**Comment by isidorn** [comment]
Issue is fixed. Adding #Closed; tag.
